From 20961b46213391bc9545c55d0214fea3abb17318 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 7 Sep 2018 09:42:38 +0300
Subject: [PATCH 4557/5242] MLK-19468: drm: imx: dcss: fix blkctl setting for
 B1 silicon

commit  1c3df0224d3ce16daf158c15af6f218ae67ec319 from
https://source.codeaurora.org/external/imx/linux-imx.git

B1 silicon has a new ID and we need to handle it properly.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-blkctl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dcss/dcss-blkctl.c b/drivers/gpu/imx/dcss/dcss-blkctl.c
index 613c349..2f13b33 100644
--- a/drivers/gpu/imx/dcss/dcss-blkctl.c
+++ b/drivers/gpu/imx/dcss/dcss-blkctl.c
@@ -116,7 +116,7 @@ int dcss_blkctl_init(struct dcss_soc *dcss, unsigned long blkctl_base)
 	if (!disp_dev || !strncmp(disp_dev, "hdmi_disp", 9))
 		blkctl->hdmi_output = true;
 
-	if (imx8_get_soc_revision() == B0_SILICON_ID)
+	if (imx8_get_soc_revision() >= B0_SILICON_ID)
 		blkctl->clk_setting = HDMI_MIPI_CLK_SEL;
 
 	dcss_blkctl_cfg(dcss);
-- 
1.7.9.5

